CONTRIBUTING.md

```diff
@@ -255,6 +255,49 @@ pub use {
 }
 ```
 
+#### Return `Self`
+
+Use `Self` as the return type instead of naming the type.
```

Comment:
**thought:** https://rust-lang.github.io/rust-clippy/master/index.html#/use_self

Reply:
We could give that lint a shot. It has an unaddressed false negative but that's not the end of the world.

Reply:
Its pretty mild but I have been using the struct name when the return is further down a function than in the same line (or close) to the top. Not sure if anyone agrees, and I can change if we use the lint. Its super minor.

Reply:
Seems its deprecated
```
warning: lint name `use_self` is deprecated and may not have an effect in the future.
```

Reply:
Also it doesn't warn for the error enum usecase, my personal favourite.

Reply:
Opened https://rust-lang.github.io/rust-clippy/master/index.html#/use_self

I use it inconsistently in my projects - I use struct names instead of `Self` in initializers. Though I should change this habit.